### DISCUSSION TITLE
Lower voltage

### DISCUSSION THREAD

**Discussion by rodriguise** [rlogiacco/BatterySense#6] [Q&A] [answered]
Would this work with 3.3v systems?

**Comment by rlogiacco** [comment, marked as answer]
Sure, you need to adapt your setup to your configuration: every time you read `5V` in the README you should consider it as `3.3V` and you are good to go.
So, if your project is powered by a LiPo (3.7-4.2V) you are in a _Higher than 5V_ context...

**Reply by yasiriqbal92** [reply]
I have a lipo battery, it is giving 3.7V. I need to make it 3.3V for esp8266 so I am using MIC29150 regulator for 3.3V. so the case is higher than 5V but I am using step down regulator. Now please tell me about R1 and R2. Is it necessary to use resistors for Sense? Can't I directly input the 3.3V from MIC29150 to sense pin? Why i need a separate voltage divider if I am already using voltage regulator (MIC29150)? Thanks in advance

**Reply by rlogiacco** [reply]
A LiPo battery outputs a voltage of `4.2V` (when fully charged) which gets lower and lower while the battery discharges. You can get down to `3.0V` without damaging the cell, but usually you want to stay on the safe side and stop draining the battery at around `3.2V`.
Now, in your case, you cannot use the whole battery because your regulator circuit will have to absorb part of the voltage due to diodes (either interna or external) which are the inherent voltage drop of the regulator (check your datasheet).
If your device has a _Vdrop_ of `0.6V` then you should stop using the battery at `3.3V` (your operating voltage) + `0.6V` (the voltage drop) = `3.9V`.

Now, your device is getting the battery voltage (which value varies) and _regulates_ it down to your requirement, keeping it steady at `3.3V`: this doesn’t allow you to measure the battery voltage and thus you cannot use it as input source for this library.
You need to know the voltage _before_ the regulator, but that is very probably beyond the input voltage range of your MCU, thus the need for the voltage divider.

**Reply by rlogiacco** [reply]
BTW, when I say _device_, I mean the `MIC29150`, when I say _MCU_ I mean your Arduino, or whatever device you are using

**Reply by yasiriqbal92** [reply]
"you should stop using the battery at 3.3V (your operating voltage) + 0.6V (the voltage drop) = 3.9V." I understand this but currently my battery is at 3.7 V so would I have to continuously measure battery output ? 
Secondly, in discharging curves, how would i know the x-axis which is current capacity? I mean do I need to see total current capacity of battery using series ammeter?

**Reply by rlogiacco** [reply]
I don't think I get your question: why are you now speaking of _current_ in terms of _electrical current_?

You want to measure your battery (at its terminals) periodically: how frequently depends on your specific requirements, the battery will get lower in voltage while getting depleted.

Regarding the discharging curves, if you are referring to the diagram I've put on the project readme, the battery capacity is measured in `mAH` (milliampere per hour) non in `mA` (milliamperes, measurable by an ammeter)... 

Please, keep the discussion focused on the library and its usage: if you need a broader set of information, StackOverflow or a good EE forum might be better.

**Reply by yasiriqbal92** [reply]
Hello, 
Sorry to question you again. I set up a voltage divider before my device (MIC29150) for 3.3V and input this (3.3v) to A0 of my MCU and I uploaded the battery monitor code. I have verified the voltage at sense pin which is 3.3V. But I am getting this 5000 value. Can you please guide me why I am not getting exact voltage? THanks 
![battery](https://user-images.githubusercontent.com/37694827/108867106-a8e6fb00-7616-11eb-8e66-c6791b4c5fa9.PNG)

**Reply by rlogiacco** [reply]
First of all, you are reading `5000` as for `5000 mV` so `5.0V` and that's because you setup your `battery` instance with value `5000`, which is wrong for your device, which is `3.3 V`... so the first value of the `begin` function should be `3300` as stated in the `README.md`, [_How To_ section](https://github.com/rlogiacco/BatterySense#how-to):

> The sense pin wiring can vary depending on your battery configuration, but here are a few examples based on the assumption you are using a 5V board: in case of a 3.3V board you should be performing the necessary adjustments.

The `begin` function call contains also another error, because you have set up your voltage divider ratio to `1`, which means _NO VOLTAGE DIVIDER_... Use the formula as per the `README.md` description, [_Voltage divider ratio_ section](https://github.com/rlogiacco/BatterySense#voltage-divider-ratio):

> The voltage divider ratio is determined by the formula `(R1 + R2) / R2`: if you use two resistors of the same value the ratio will be **2**, which can be interpreted as _whatever value we read it will be half of the actual value_. This allows us to sense batteries up to `10V`. If you use a `22k Ohm` resistor for `R1` and a `10k Ohm` for `R2` than your voltage ratio will be `3.2` and you will be able to safely monitor a `12-15V` battery. 

Please note all the examples refer to a `5V` device, **so you have to adjust your calculations** to a `3.3V` one: you can measure batteries up to `6.6V` with a divider ratio of `2` and up to `10.5V` with a divider ratio of `3.2`!

Finally, if you need further assistance, please post a schematics of your circuit: you can use [this website](https://www.falstad.com/circuit/circuitjs.html) (which also has a simulator) or [Upverter ](https://upverter.com/) or any other tool you prefer.

Hope this helps!
